### ISSUE TITLE
Rule no-sequences is misleading when different comma is left out

### ISSUE THREAD

**Issue by FredyC** [eslint/eslint#6652]
**What version are you using?**
Version 2.11.1

**What did you do?**
```js
const schema = createSchema([
	['doFindAirports',
		['location', AV('location')]
		['locale', AV('locale')],
	],
	['findAirportsSuccess', AV('airports')],
	['findAirportsFailed'],
]);
```
**What happened?**
_Unexpected use of comma operator. (no-sequences), Line 4, Column 1_

**What did you expect to happen?**
I did forgot to add comma after the line `['location', AV('location')]` and got this misleading error messages telling me wrongly I cannot use comma on line 4: `['locale', <--`. Adding missing comma fixes the problem, but it's kinda hard to spot since it is showing me different unrelated error.

I am not entirely sure if there is a rule that covers missing commas in arrays like this, but it should have a priority in that case.

Here goes my config
```yml
extends:
  eslint-config-airbnb

parser:
  babel-eslint

plugins:
  - babel
  - react

settings:
  ecmascript: 6
  import/resolver: webpack

ecmaFeatures:
  jsx: true
  modules: true
  destructuring: true
  classes: true
  forOf: true
  blockBindings: true
  arrowFunctions: true

env:
  browser: true

rules:
  indent: [error, tab]
  object-curly-spacing: off
  yield-star-spacing: off
  no-use-before-define: off
  no-unused-vars: warn
  new-cap: off
  quotes: [warn, single, {
    avoidEscape: true, allowTemplateLiterals: true
  }]

  #
  # eslint-plugin-babel
  #
  babel/generator-star-spacing: [warn, before]
  babel/array-bracket-spacing: warn
  babel/object-curly-spacing: [error, always, {
    objectsInObjects: false, arraysInObjects: false
  }]
  babel/object-shorthand: [2, always]
  babel/arrow-parens: [warn, always]
  babel/no-await-in-loop: warn

  react/react-in-jsx-scope: off
  react/jsx-indent: [warn, tab]
  react/jsx-indent-props: [warn, tab]
```

**Comment by vsemozhetbyt** [comment]
It seems like Eslint sees the expression

`['location', AV('location')]['locale', AV('locale')]`

as an array `['location', AV('location')]` and immediately added subscript operator (property accessor) `['locale', AV('locale')]`. And this is a valid code. However, in the subscript operator Eslint sees two expresions devided by comma: `'locale'` and `AV('locale')` as if you call a string, then throw off the result, then call `AV('locale')` and use its result as an calculated property name. And this is just the valid forbidden case of the rule.

**Comment by mysticatea** [comment]
Thank you for this issue.

@vsemozhetbyt is right.

```js
const schema = createSchema([
    ['doFindAirports',
        ['location', AV('location')]    // ← it needs a comma here. ★
        ['locale', AV('locale')],
    ],
    ['findAirportsSuccess', AV('airports')],
    ['findAirportsFailed'],
]);
```

`'locale', AV('locale')` is a SequenceExpression since the comma of ★ is lacking.
I'd recommend a use of [no-unexpected-multiline](http://eslint.org/docs/rules/no-unexpected-multiline) rule. That rule would show better warnings for this kind of mistakes.

    Unexpected newline between object and [ of property access. (no-unexpected-multiline)

**Comment by FredyC** [comment]
Thanks for a quick reply. I confirm that by enabling `no-unexpected-multiline` it's slightly better behavior for finding a problem and it did not influenced my current code, so it's a completely viable solution.
